src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -449,10 +459,21 @@ private BulkResponse execute() throws Exception {
                   System.currentTimeMillis() - startTime
               );
             }
+            batch.forEach(r -> recordMap.remove(r));
             return bulkRsp;
           } else if (responseContainsMalformedDocError(bulkRsp)) {
             retriable = bulkRsp.isRetriable();
             handleMalformedDoc(bulkRsp);
+            for (R record : batch) {
+              SinkRecord original = recordMap.get(record);
+              if (original != null && reporter != null) {
```

Comment:
I read this and I'm thinking: should we make more things conditional to the `reporter` being not `null`?
At least this whole `for` loop does not need to run unless `reporter != null`, right?